test/e2e/basic/util.go

```diff
@@ -286,3 +286,17 @@ func WaitForClusterOperatorCondition(cs *framework.ClientSet, interval, duration
 	}
 	return nil
 }
+
+// IsNodeReady helps determine if a given node is ready or not.
+func IsNodeReady(node corev1.Node) bool {
+	found := false
+	for _, condition := range node.Status.Conditions {
```

Comment:
Is there a reason why we need to iterate over all the conditions even when we find `corev1.NodeReady`?  Would the following work?
```
// IsNodeReady helps determine if a given node is ready or not.
func IsNodeReady(node coreapi.Node) bool {
        for _, condition := range node.Status.Conditions {
                if condition.Type == coreapi.NodeReady {
                        return condition.Status == coreapi.ConditionTrue
                }
        }
        return false
}

```

Reply:
Thanks Jiri, good catch. This looks to me to be totally equivalent but with less iterations.